transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/internal/wizards/XMLPage.java

```diff
@@ -331,7 +331,7 @@ private String selectResourceFromWorkspace(Shell shell, final String extension)
         }
         ClasspathResourceSelectionDialog dialog = null;
         HashSet<String> extensions = new HashSet<>();
-        extensions.add("xml"); //$NON-NLS-1$
+        if (!"xsd".equals(extension)) extensions.add("xml"); //$NON-NLS-1$
```

Comment:
I think it would make sense to create constants for those values. (xml, xsd, wsdl)